src/Compiler/CompilerOpenFPGA.cpp

```diff
@@ -729,6 +732,29 @@ bool CompilerOpenFPGA::DesignChanged(
 }
 
 bool CompilerOpenFPGA::Analyze() {
+  auto printTopModules = [](const std::filesystem::path& filePath,
+                            std::ostream* out) {
+    // Check for "topModule" in a given json filePath
+    // Assumed json format is [ { "topModule" : "some_value"} ]
+    if (out) {
+      if (FileUtils::FileExists(filePath)) {
+        std::ifstream file(filePath);
+        json data = json::parse(file);
+        if (data.is_array()) {
+          std::vector<std::string> topModules;
+          std::transform(data.begin(), data.end(),
+                         std::back_inserter(topModules),
+                         [](json val) -> std::string {
+                           return val.value("topModule", "");
```

Comment:
In case of multiple top modules, we still have one key with multiple values, correct?

Reply:
No that's not how I understand it from Alain's reply. I originally was reading the json assuming there was only 1 element in the array. Now the code will step through multiple objects in the top array of the json and grab "topModule" from each object. If that's not the case and there is only 1 object in an unnecessary array and "topModule" contains space separated modules, that should still work as my code will read the first array value which in that case would just have multiple stored in it. Only difference in that scenario is the values would display space separated instead of comma separated which is fine since this is visual, user level info that doesn't require a parseable formatting.